python-clusters/create-eks-cluster/cluster.json

```diff
@@ -66,6 +66,45 @@
              "mandatory" : true,
              "defaultValue" : true
         },
+        {
+            "name": "privateCluster",
+            "label": "Fully-private",
+            "type": "BOOLEAN",
+            "description": "VPC endpoints will be created for the cluster. The first security group of the Network Settings will be used as shared SG."
+        },
+        {
+            "name": "skipEndpointCreation",
+            "label": "Skip VPC Endpoints creation",
+            "type": "BOOLEAN",
+            "description": "If the VPC endpoints already exist, don't attempt to create new ones",
+            "visibilityCondition": "model.privateCluster == true"
+        },
+        {
+            "name": "controlPlaneSG",
+            "label": "Control plane SG",
+            "type": "STRING",
+            "description": "Security group to use for the control plane. Can be used to allow access from the DSS VM.",
+            "visibilityCondition": "model.privateCluster == true"
+        },
+        {
+            "name": "sharedSG",
+            "label": "Shared SG",
+            "type": "STRING",
+            "description": "Security group to use for control plane and VPC endpoints. Can be used to allow access from the DSS VM.",
+            "visibilityCondition": "model.privateCluster == true"
+        },
+        {
+            "name": "injectedSG",
+            "label": "Injected DSS SG",
+            "type": "STRING",
+            "description": "Security group to add as inbound on the cluster's autocreated SGs. Should allow access from the DSS VM."
+        },
```

Comment:
sharedSG is a SG that eksctl with smother on all the cluster and endpoints, so if it's a SG of your DSS VM, that automatically puts your VM in the club and it can talk to all the endpoints. Otherwise you need to add your VM in the club, which can be done by adding the (private) IP of the VM to the SG of the cluster and endpoints, or by adding a specific SG as allowed source. The latter is done with this `injectedSG` : pass a SG of your DSS VM, and it becomes an allowed source of traffic for the endpoints